From 40b613891def469788ace0860d9eecfbfb05fab8 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Thu, 7 Apr 2016 11:10:07 +0300
Subject: [PATCH 0157/1345] dts: Add AP806-Z1 specific pic dts information

commit  2b4b719e67959ebb6e1a342c7de96f6819a4048e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

interrupt mask polarity and gic interrupt index are different between
AP806-Z1 and AP806-A0.

Change-Id: If3e77b34d7b670966d6c00356d65d6dff4ef493d
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28870
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-ap806-z1-quad.dtsi     |    6 ++++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      |    1 +
 2 files changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
index 4a10541..f685bf6 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
@@ -82,6 +82,12 @@
 
 	ap806 {
 		config-space@f0000000 {
+			pic: interrupt-controller@3f0100 {
+				/* Override A0 pic settings to match Z1 spec. */
+				int-en-pol = <1>;
+				interrupts = <GIC_PPI 12 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
 			/* For SPI, I2C, and Serial units override the clock */
 			spi0: spi@510600 {
 				clocks = <&ringclk 2>;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 176df75..b6fe7a9 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -165,6 +165,7 @@
 			pic: interrupt-controller@3f0100 {
 				compatible = "marvell,armada-8k-pic";
 				reg = <0x3f0100 0x10>;
+				int-en-pol = <0>;
 				#interrupt-cells = <1>;
 				interrupt-controller;
 				interrupts = <GIC_PPI 15 IRQ_TYPE_LEVEL_HIGH>;
-- 
1.7.9.5

